pkg/errs/errors.go

```diff
@@ -0,0 +1,47 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package errs
+
+import (
+	"errors"
+
+	"github.com/containerd/errdefs"
+)
+
+var (
```

Comment:
Many of these are already defined in https://github.com/containerd/errdefs/blob/02b65bcc1b50358f7c1cca419c76f538349af784/errors.go#L37

`ErrInvalidArgument`, `ErrFailedPrecondition`.

And `ErrSystemIsBroken`, `ErrNetworkCondition`, `ErrServerIsMisbehaving` are similar to `ErrInternal` and `ErrUnavailable`.

Can we reuse the definition in containerd/errdefs?

Reply:
I can alias them if you think it is helpful, but I think it is a bad idea to just use them directly as outlined above.

Reply:
If you think it's a bad idea, maybe just wrap it with %w so it works for both nerdctl's and containerd's definition.

Reply:
> If you think it's a bad idea, maybe just wrap it with %w so it works for both nerdctl's and containerd's definition.

Just to clarify: containerd returning errors from errdefs are fine - and may of course be handled and wrapped, and should NOT be removed, or ignored - and none of this proposal will "break" existing code testing for `errdefs.ErrNotFound`.

example:
```
if errors.Is(containerdErr, errdefs.ErrNotFound) {
    return errors.Join(ErrVolumeRemovalFail, containerdErr)
}
```

Same applies to any other error - from any third-party lib, or from golang.